dev/archery/archery/integration/datagen.py

```diff
@@ -743,6 +763,82 @@ class LargeStringColumn(_BaseStringColumn, _LargeOffsetsMixin):
     pass
 
 
+class BinaryViewColumn(PrimitiveColumn):
+
+    def _encode_value(self, x):
+        return frombytes(binascii.hexlify(x).upper())
+
+    def _get_buffers(self):
+        char_buffers = []
+        # a small default char buffer size is used so we get multiple
+        # character buffers without massive arrays
+        DEFAULT_BUFFER_SIZE = 32
+        INLINE_SIZE = 12
+
+        data = []
+        for i, v in enumerate(self.values):
+            if not self.is_valid[i]:
+                v = b''
+            assert isinstance(v, bytes)
+
+            if len(v) > INLINE_SIZE:
+                offset = 0
+                if len(v) > DEFAULT_BUFFER_SIZE:
```

Comment:
This is probably redundant with the third condition below (`len(char_buffers[-1]) + len(v) > DEFAULT_BUFFER_SIZE`)

Reply:
I'll try to reorganize